ambry-router/src/test/java/com.github.ambry.router/GetManagerTest.java

```diff
@@ -118,6 +120,46 @@ public void testCompositeBlobGetSuccess()
   }
 
   /**
+   * Test that a bad user defined callback will not crash the router.
+   * @throws Exception
+   */
+  @Test
+  public void testBadCallback()
+      throws Exception {
+    router = getNonBlockingRouter();
+    setOperationParams(chunkSize * 6 + 11);
+    final CountDownLatch callbackCalled = new CountDownLatch(1);
+    String blobId = router.putBlob(putBlobProperties, putUserMetadata, putChannel).get();
+    router.getBlobInfo(blobId, new Callback<BlobInfo>() {
+      @Override
+      public void onCompletion(BlobInfo result, Exception exception) {
+        callbackCalled.countDown();
+        throw new RuntimeException("Throwing an exception in the user callback");
+      }
+    }).get();
+    router.getBlob(blobId, new Callback<ReadableStreamChannel>() {
+      @Override
+      public void onCompletion(ReadableStreamChannel result, Exception exception) {
+        callbackCalled.countDown();
+        throw new RuntimeException("Throwing an exception in the user callback");
+      }
+    }).get();
+    Assert.assertTrue("Callback not called.", callbackCalled.await(2, TimeUnit.SECONDS));
```

Comment:
would be great if we could also guard against callbacks getting called more than once, with this new change.